cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/analyzer/goals/ResourceDistributionGoal.java

```diff
@@ -321,22 +389,180 @@ protected boolean rebalanceByMovingLoadIn(Broker broker,
     return true;
   }
 
-  protected boolean rebalanceByMovingLoadOut(Broker broker,
+  private SortedSet<Replica> replicasToSwapWith(Broker broker,
+                                                Set<String> excludedTopics,
+                                                double swapLoadLimit,
+                                                boolean isSwapIn) {
+    SortedSet<Replica> candidateReplicasToSwapWith = new TreeSet<>((r1, r2) -> {
+      int result = isSwapIn ? Double.compare(r2.load().expectedUtilizationFor(resource()), r1.load().expectedUtilizationFor(resource()))
+                            : Double.compare(r1.load().expectedUtilizationFor(resource()), r2.load().expectedUtilizationFor(resource()));
+      return result == 0 ? r1.topicPartition().toString().compareTo(r2.topicPartition().toString()) : result;
+    });
+    candidateReplicasToSwapWith.addAll(
+        resource() == Resource.NW_OUT
```

Comment:
This is probably the time to use `if...else...` rather than the seemingly short question mark.